src/lib/device-online-state.ts

```diff
@@ -0,0 +1,337 @@
+import * as Bluebird from 'bluebird';
+import { DEFAULT_SUPERVISOR_POLL_INTERVAL } from './env-vars';
+import { noop } from 'lodash';
+import { resinApi, root } from '../platform';
+import { captureException } from '../platform/errors';
+import {
+	createPromisifedRedisClient,
+	PromisifedRedisClient,
+} from './redis-promise';
+import * as RedisSMQ from 'rsmq';
+import {
+	REDIS_HOST,
+	REDIS_PORT,
+	API_HEARTBEAT_STATE_ACTIVE,
+	API_HEARTBEAT_STATE_TIMEOUT_SECONDS,
+} from './config';
+
+export const getPollIntervalForDevice = (uuid: string): Bluebird<number> => {
+	return resinApi
```

Comment:
Can you extract these queries to prepared queries and use parameter aliases please (see routes/devices.ts for examples).  It'll allow them to be handled more efficiently internally (and they should be called regularly enough for that to matter)